src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs

```diff
@@ -109,6 +124,28 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
 
                 output.Attributes.SetAttribute(SrcAttributeName, _fileVersionProvider.AddFileVersionToPath(Src));
             }
+
+            //Retrieve any existing onerror handler code
+            var onError = output.Attributes[OnErrorAttributeName]?.Value as string;
+
+            //Check if there's a fallback source and no onerror handler
+            if (!string.IsNullOrWhiteSpace(FallbackSrc) && string.IsNullOrWhiteSpace(onError))
+            {
+                string resolvedUrl;
+                if (TryResolveUrl(FallbackSrc, out resolvedUrl))
+                {
+                    FallbackSrc = resolvedUrl;
+                }
+
+                if (AppendVersion)
+                {
+                    FallbackSrc = _fileVersionProvider.AddFileVersionToPath(FallbackSrc);
+                }
+
+                //Apply fallback handler code
+                onError = $"this.src = '{FallbackSrc}'";
```

Comment:
According to this SO response, https://stackoverflow.com/a/980948, you'll need to also account for FallbackSrc erroring, so this needs to be a bit more elaborate:

` <img src="foo.jpg" onerror="if (this.src != 'error.jog') this.src = 'error.jpg';">`

More importantly, `onerror` is an inline handler and has the same sort of issues with CSP that are described here - https://github.com/aspnet/Mvc/issues/4300. IMO, it would make sense to figure out a strategy for `ScriptTagHelper` that we could apply here.

@DamianEdwards thoughts?